### PR TITLE
Remove the clippy::result_large_err clippy exceptions 

### LINKED ISSUE #1897: Remove the clippy::result_large_err clippy exceptions
**Issue summary**

Possible options include making the error types smaller or using `Box`. No additional error casting should be required.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1897

**Other information and links**

### CODE REVIEW

File: blockchain/chain_sync/src/validation.rs [hunk 1]
```diff
@@ -75,65 +75,73 @@ impl<'a> TipsetValidator<'a> {
...
-    ) -> Result<Cid, TipsetValidationError> {
+    ) -> Result<Cid, Box<TipsetValidationError>> {
         // Generate message CIDs
         let bls_cids = bls_msgs
             .iter()
             .map(Cbor::cid)
-            .collect::<Result<Vec<Cid>, fvm_ipld_encoding::Error>>()?;
+            .collect::<Result<Vec<Cid>, fvm_ipld_encoding::Error>>()
+            .map_err(Into::into)
+            .map_err(Box::new)?;
```
Comment (lemmih) [inline blockchain/chain_sync/src/validation.rs:130]: Can't you add something like:
```rust
impl From<EncodingError> for Box<TipsetValidationError> {
    fn from(err: EncodingError) -> Self {
        Box::new(TipsetValidationError::Encoding(err))
    }
}
```

And then the conversion happens automatically?

Comment (tyshko5) [reply blockchain/chain_sync/src/validation.rs]: Done

File: blockchain/consensus/deleg_cns/src/consensus.rs [hunk 2]
```diff
@@ -136,6 +136,7 @@ impl Consensus for DelegatedConsensus {
     {
         crate::validation::validate_block(&self.chosen_one, state_manager, block)
             .await
+            .map_err(|err| *err)
```
Comment (lemmih) [inline blockchain/consensus/deleg_cns/src/consensus.rs:139]: How about changing the `Error` type to `Box<DelegatedConsensusError>`?

File: blockchain/consensus/deleg_cns/src/validation.rs [hunk 3]
```diff
@@ -22,13 +22,17 @@ pub(crate) async fn validate_block<DB: BlockStore + Sync + Send + 'static>(
...
     let header = block.header();
 
     block_sanity_checks(header)?;
 
-    let base_tipset = chain_store.tipset_from_keys(header.parents()).await?;
+    let base_tipset = chain_store
+        .tipset_from_keys(header.parents())
+        .await
+        .map_err(Into::into)
+        .map_err(Box::new)?;
```
Comment (lemmih) [inline blockchain/consensus/deleg_cns/src/validation.rs:35]: Add a `From` impl to take care of this?

File: blockchain/consensus/deleg_cns/src/validation.rs [hunk 4]
```diff
@@ -101,14 +102,16 @@ where
     // Workaround for the bug where Forest strips the network type from the Address
     // and then puts back always the mainnet variant, so the `t` prefix becomes `f`.
     let chosen_addr = state_manager
-        .lookup_id(chosen_one, base_tipset)?
+        .lookup_id(chosen_one, base_tipset)
+        .map_err(Into::into)
+        .map_err(Box::new)?
```
Comment (lemmih) [inline blockchain/consensus/deleg_cns/src/validation.rs:107]: Another `From` impl?

File: blockchain/consensus/deleg_cns/src/validation.rs (hunk 3 above)
Comment (tyshko5) [reply blockchain/consensus/deleg_cns/src/validation.rs]: Done

File: blockchain/chain_sync/src/chain_muxer.rs [hunk 5]
```diff
@@ -462,7 +462,7 @@ where
                 "Validating tipset received through GossipSub failed: {}",
                 why
             );
-            return Err(why.into());
+            return Err((*why).into());
```
Comment (lemmih) [inline blockchain/chain_sync/src/chain_muxer.rs:465]: You can probably box the outer error and change the `From` impl.

Comment (tyshko5) [reply blockchain/chain_sync/src/chain_muxer.rs]: Solved a bit differently

File: blockchain/chain_sync/src/chain_muxer.rs [hunk 6]
```diff
@@ -51,7 +51,7 @@ pub enum ChainMuxerError<C: Consensus> {
     #[error("Tipset range syncer error: {0}")]
     TipsetRangeSyncer(#[from] TipsetRangeSyncerError<C>),
     #[error("Tipset validation error: {0}")]
-    TipsetValidator(#[from] TipsetValidationError),
+    TipsetValidator(#[from] Box<TipsetValidationError>),
```
Comment (lemmih) [inline blockchain/chain_sync/src/chain_muxer.rs:54]: It's slightly peculiar to only box one thing but I think it's fine in this case.

File: blockchain/chain_sync/src/chain_muxer.rs [hunk 7]
```diff
@@ -176,7 +176,7 @@ where
         tipset_sender: Sender<Arc<Tipset>>,
         tipset_receiver: Receiver<Arc<Tipset>>,
         cfg: SyncConfig,
-    ) -> Result<Self, ChainMuxerError<C>> {
+    ) -> Result<Self, Box<ChainMuxerError<C>>> {
```
Comment (lemmih) [inline blockchain/chain_sync/src/chain_muxer.rs:179]: This is still required to box `ChainMuxerError` now that you've boxed `TipsetValidationError`?

File: blockchain/consensus/deleg_cns/src/validation.rs [hunk 8]
```diff
@@ -11,6 +11,18 @@ use fvm_shared::address::Address;
...
+        Box::new(Into::into(err))
+    }
+}
+
+impl From<forest_state_manager::Error> for Box<DelegatedConsensusError> {
+    fn from(err: forest_state_manager::Error) -> Self {
+        Box::new(Into::into(err))
+    }
+}
+
```
Comment (lemmih) [inline blockchain/consensus/deleg_cns/src/validation.rs:25]: Maybe move these to `blockchain/consensus/deleg_cns/src/consensus.rs` where `DelegatedConsensusError` is defined (and where the other From impls are implicitly defined).

**Review by lemmih — CHANGES_REQUESTED** [review]
Looks good. There's one unnecessary `map_err`, though.

File: blockchain/chain_sync/src/chain_muxer.rs (hunk 7 above)
Comment (tyshko5) [reply blockchain/chain_sync/src/chain_muxer.rs]: Removed

File: blockchain/consensus/deleg_cns/src/validation.rs (hunk 8 above)
Comment (tyshko5) [reply blockchain/consensus/deleg_cns/src/validation.rs]: Moved

File: blockchain/consensus/deleg_cns/src/consensus.rs (hunk 2 above)
Comment (tyshko5) [reply blockchain/consensus/deleg_cns/src/consensus.rs]: Changed

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**